shell/platform/android/test/io/flutter/embedding/android/FlutterViewTest.java

```diff
@@ -480,6 +480,68 @@ public void systemInsetGetInsetsFullscreenLegacy() {
     assertEquals(103, viewportMetricsCaptor.getValue().paddingRight);
   }
 
+  // This test uses the API 30+ Algorithm for window insets. The legacy algorithm is
+  // set to -1 values, so it is clear if the wrong algorithm is used.
+  @Test
+  @TargetApi(30)
+  @Config(sdk = 30)
+  public void systemInsetDisplayCutoutSimple() {
+    RuntimeEnvironment.setQualifiers("+land");
+    FlutterView flutterView = spy(new FlutterView(RuntimeEnvironment.systemContext));
+    ShadowDisplay display =
+        Shadows.shadowOf(
+            ((WindowManager)
+                    RuntimeEnvironment.systemContext.getSystemService(Context.WINDOW_SERVICE))
+                .getDefaultDisplay());
+    // display.setRotation(0);
+    assertEquals(0, flutterView.getSystemUiVisibility());
+    when(flutterView.getWindowSystemUiVisibility()).thenReturn(0);
+    when(flutterView.getContext()).thenReturn(RuntimeEnvironment.systemContext);
+
+    FlutterEngine flutterEngine =
+        spy(new FlutterEngine(RuntimeEnvironment.application, mockFlutterLoader, mockFlutterJni));
```

Comment:
ultra nit: if you can figure out the chain needed to just mock instead of spy, it might make this test more stable in the long run. Might not be possible though.